app/Http/Controllers/Playlist/PlaylistsController.php

```diff
@@ -727,17 +745,23 @@ public function storeItem(Request $request, $playlist_id)
             ->where('id', $playlist_id)->first();
 
         if (!$playlist) {
-            return $this->setStatusCode(404)->replyWithError('Playlist Not Found');
+            return $this->setStatusCode(SymfonyResponse::HTTP_NOT_FOUND)->replyWithError('Playlist Not Found');
         }
 
         $playlist_items = json_decode($request->getContent());
         $single_item = checkParam('fileset_id');
 
+        if (!$single_item && !is_array($playlist_items)) {
```

Comment:
Added the constraint to validate if fileset_id is empty and it is a single item. It should return a bad request response because the fileset_id is required to create a new playlist item.